hazelcast/include/hazelcast/client/imap.h

```diff
@@ -603,9 +603,8 @@ namespace hazelcast {
                                                                                                 entry.second);
                                                                                         // it is guaranteed that all values are non-null
                                                                                         assert(val.has_value());
-                                                                                        result[to_object<K>(
-                                                                                                entry.first).value()] = std::move(
-                                                                                                val.value());
+                                                                                        result.emplace(to_object<K>(
+                                                                                                entry.first).value(), val.value());
```

Comment:
This change will introduce an additional copy of the value, right? Can't we still move the value, so the move ctor is called instead of the copy ctor?
```suggestion
                                                                                        result.emplace(to_object<K>(
                                                                                                entry.first).value(), std::move(val.value()));
```

Reply:
Yes, you're absolutely right, thank you for the suggestion!

Reply:
Thanks!